### ISSUE TITLE
`truncate_dim` on `BertModel`

### ISSUE THREAD

**Issue by kerem0comert** [huggingface/transformers#32881]
### Feature request

I have a pipeline to finetune an instance of `BertModel`, on a `text-classification` task. 
I would like to use [this new embedding model](https://huggingface.co/aari1995/German_Semantic_V3#usage) as my base embedding now.
As can be seen in the example they provide, they are able to pass different values for `matryoshka_dim` into the `SentenceTransformer` instance through the `truncate_dim` argument.
However, I was not able to do this on the `BertModel` in the following code snippet that I have in my code:

```python
 self.bert_backbone = BertModel.from_pretrained(
            pretrained_model_name_or_path=self.config.embedding_model_file.model_name,
            cache_dir=Path(self.config.embedding_model_file.cache_dir),
        ).to(self.device)
```

And I do not want to use a `SentenceTransformer` instance either as in my training loop I would like to be able to get:

```python
bert_outputs: BaseModelOutputWithPoolingAndCrossAttentions = (
                    self.bert_backbone(
                        input_ids=input_ids, attention_mask=attention_mask
                    )
                )
                bert_logits: Tensor = bert_outputs.last_hidden_state[
                    :, 0, :
                ]  # Take the [CLS] token output
```

and I am not sure if this code would work also with a simple swap to `SentenceTransformer`. In any case, I think that this is a potential parameter that `BertModel` should support, and maybe it does but I am just missing it.

Thanks in advance!

### Motivation

To be able to extend the `BertModel` further

### Your contribution

.

**Comment by amyeroberts** [comment]
Hi, thanks for raising an issue! 

As this involves custom code, this is a question best placed in our [forums](https://discuss.huggingface.co/). We try to reserve the github issues for feature requests and bug reports.

**Comment by kerem0comert** [comment]
Alright, [did so here for reference](https://discuss.huggingface.co/t/truncate-dim-on-bertmodel/103110)

**Comment by amyeroberts** [comment]
Technically yes, although for it to be a feature request the exact behaviour and feature added should be clear and well specified. 

Half of the issue here is a question about how to do and approach a custom use case. If after discussion in the forum you realise there's a specific, singular feature that you think should be added (which improves the library as a whole, rather than enables a single use-case) then we'd be very happy to hear the request!
